ngraph/core/src/op/squeeze.cpp

```diff
@@ -149,28 +149,44 @@ shared_ptr<Node> op::Squeeze::clone_with_new_inputs(const OutputVector& new_args
 namespace squeeze
 {
     template <element::Type_t ET>
-    bool evaluate(const HostTensorPtr& arg0, const HostTensorPtr& out)
+    bool evaluate(const HostTensorPtr& arg0, const HostTensorPtr& arg1, const HostTensorPtr& out)
     {
-        runtime::reference::copy(
-            arg0->get_data_ptr<ET>(), out->get_data_ptr<ET>(), shape_size(out->get_shape()));
+        const auto data_rank = arg0->get_partial_shape().rank().get_length();
+        const auto axes_num = shape_size(arg1->get_shape());
+        auto norm_axes = normalize_axes(
+            "",
+            std::vector<int64_t>(arg1->get_data_ptr<ET>(), arg1->get_data_ptr<ET>() + axes_num),
+            data_rank);
+        set<size_t, greater<size_t>> ordered_axes(norm_axes.begin(), norm_axes.end());
+
+        auto out_shape = arg0->get_shape();
+        for (const auto& axis : ordered_axes)
+        {
+            if (out_shape[axis] != 1)
+            {
+                throw ngraph_error("Squeeze dimension is not equal to 1");
+            }
+            out_shape.erase(out_shape.begin() + axis);
```

Comment:
@AndrewBakalinIntel, did you check it in case several axes for squeezing? To me it looks that after the first squeeze the other axes invalidates. Please correct me if I am wrong. Also please add tests to cover it.

Reply:
I use ordered set with `greater` comparing function which means that axes in `ordered_axes` are in descending order, so here I do erasing from end to the beginning and no invalidation happens.
We at least have tests with multiple axes as layers' tests on CPU and GPU and they are using `evaluate()` method, so it's already been covered.